pkg/controller/core/workload_controller.go

```diff
@@ -287,20 +290,23 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	lqActive := ptr.Deref(lq.Spec.StopPolicy, kueue.None) == kueue.None
 	if lqExists && lqActive && isDisabledRequeuedByLocalQueueStopped(&wl) {
 		workload.SetRequeuedCondition(&wl, kueue.WorkloadLocalQueueRestarted, "The LocalQueue was restarted after being stopped", true)
-		return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock)
+		err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 	if cqOk {
 		// because we need to react to API cluster cq events, the list of checks from a cache can lead to race conditions
 		cq := kueue.ClusterQueue{}
-		if err := r.client.Get(ctx, types.NamespacedName{Name: string(cqName)}, &cq); err != nil {
+		err := r.client.Get(ctx, types.NamespacedName{Name: string(cqName)}, &cq)
+		if client.IgnoreNotFound(err) != nil {
 			return ctrl.Result{}, err
```

Comment:
```suggestion
		if err := r.client.Get(ctx, types.NamespacedName{Name: string(cqName)}, &cq); err != nil {
			return ctrl.Result{}, client.IgnoreNotFound(err)
		}
```

This also leads to an unexpected following steps.